tket/test/src/Circuit/test_Boxes.cpp

```diff
@@ -811,6 +811,24 @@ SCENARIO("QControlBox", "[boxes]") {
     std::optional<Eigen::MatrixXcd> box_u = qcbox.get_box_unitary();
     REQUIRE(V.isApprox(box_u.value()));
   }
+  GIVEN("Random unitary box controlled by state") {
+    Eigen::MatrixXcd u = random_unitary(8, 1);
+    Unitary3qBox ubox(u);
+    Op_ptr op = std::make_shared<Unitary3qBox>(ubox);
+    QControlBox qcbox(op, 3, {1, 0, 1});
+    std::shared_ptr<Circuit> c = qcbox.to_circuit();
+    const Eigen::MatrixXcd circ_u = tket_sim::get_unitary(*c);
+    Eigen::MatrixXcd V = Eigen::MatrixXcd::Identity(64, 64);
+    for (unsigned i = 0; i < 8; i++) {
+      for (unsigned j = 0; j < 8; j++) {
+        V(40 + i, 40 + j) = u(i, j);
```

Comment:
```suggestion
    QControlBox qcbox(op, 3, {1, 0, 0});
    std::shared_ptr<Circuit> c = qcbox.to_circuit();
    const Eigen::MatrixXcd circ_u = tket_sim::get_unitary(*c);
    Eigen::MatrixXcd V = Eigen::MatrixXcd::Identity(64, 64);
    for (unsigned i = 0; i < 8; i++) {
      for (unsigned j = 0; j < 8; j++) {
        V(32 + i, 32 + j) = u(i, j);
```
I prefer this test because it catches endianness mistakes.